ruby/deepgram-language-detection/index.rb

```diff
@@ -0,0 +1,39 @@
+require "uri"
+require "net/http"
+require "openssl"
+require "json"
+
+
+def main(req, res)
+  fileUrl = nil
+  begin
+    payload = JSON.parse(req.payload)
+    fileUrl = payload["fileUrl"]
+  rescue Exception => err
+      puts err
+      raise "Payload is invalid."
+  end
+  url = URI("https://api.deepgram.com/v1/listen?detect_language=true&punctuate=true")
+  http = Net::HTTP.new(url.host, url.port)
+  http.use_ssl = true
+  http.verify_mode = OpenSSL::SSL::VERIFY_NONE
+
+  request = Net::HTTP::Post.new(url)
+  request["content-type"] = "application/json"
+  request["Authorization"] = "Token #{req.env["DEEPGRAM_API_KEY"]}"
```

Comment:
Make sure to use `req.variables`:

```suggestion
  request["Authorization"] = "Token #{req.variables["DEEPGRAM_API_KEY"]}"
```